src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -728,15 +744,14 @@ export class GuidedTourService {
      * @param guidedTour that should be enabled
      */
     public enableTourForCourseExerciseComponent(course: Course | null, guidedTour: GuidedTour): Exercise | null {
-        if (!guidedTour.exerciseShortName || !course || !course.exercises) {
+        if (!course || !course.exercises || !this.guidedTourMapping) {
             return null;
         }
-        const exerciseForGuidedTourExists = course.exercises.find(
-            exercise => (exercise.type === ExerciseType.PROGRAMMING && exercise.shortName === guidedTour.exerciseShortName) || exercise.title === guidedTour.exerciseShortName,
-        );
-        if (exerciseForGuidedTourExists) {
+
+        const exerciseForGuidedTour = course.exercises.find(exercise => this.isGuidedTourAvailableForExercise(exercise, guidedTour));
+        if (this.isGuidedTourAvailableForCourse(course) && exerciseForGuidedTour) {
```

Comment:
I think it is better to first execute `isGuidedTourAvailableForCourse` and only then find the exercise